barretenberg/cpp/src/barretenberg/vm/avm/trace/trace.cpp

```diff
@@ -2357,45 +2357,71 @@ void AvmTraceBuilder::op_l1_to_l2_msg_exists(uint8_t indirect,
     debug("l1_to_l2_msg_exists side-effect cnt: ", side_effect_counter);
 }
 
-void AvmTraceBuilder::op_get_contract_instance(uint8_t indirect, uint32_t address_offset, uint32_t dst_offset)
+void AvmTraceBuilder::op_get_contract_instance(
+    uint8_t indirect, uint8_t member_enum, uint16_t address_offset, uint16_t dst_offset, uint16_t exists_offset)
 {
+    ASSERT(member_enum < static_cast<int>(ContractInstanceMember::MAX_MEMBER));
```

Comment:
Ultimately, we will have to handle and bytecode. Therefore, an assert is not the longterm solution, we will also have to properly constrain this in the circuit and raise an error. At least, add a TODO.

Reply:
I'll change it to an if-statement that raises the error flag & a comment to constrain it with a range check on the enum operand